materialize-bigquery/bigquery.go

```diff
@@ -187,16 +197,9 @@ func newBigQueryDriver() *sql.Driver {
 
 func prereqs(ctx context.Context, ep *sql.Endpoint) *sql.PrereqErr {
 	cfg := ep.Config.(*config)
+	client := ep.Client.(*client)
```

Comment:
How come we don't need to call `cfg.client` anymore? `cfg.client` seems to be doing a lot of stuff 🤔

Reply:
It's already part of the endpoint, initialized here: https://github.com/estuary/connectors/blob/5afc5133a39a9ac989d63f36f5265825a4c88650/materialize-sql/driver.go#L77C11-L79

The genesis of these invalid-credentials-json errors was kind of confusing before, since we were doing a check here for exactly that, but it wasn't being hit. And that was because the client already gets created as part of building the endpoint that gets passed to here.

That said, I think this structure could be improved, and at some point in the future I'd like to look at making the `prereqs` checks part of the `Client` interface.